src/Jackett.Common/Indexers/InternetArchive.cs

```diff
@@ -166,29 +165,26 @@ private ReleaseInfo MakeRelease(JToken torrent)
         {
             var release = new ReleaseInfo();
 
-            var id = (string) torrent["identifier"];
-            var title = torrent["title"] is JArray ?
-                (string)((JArray)torrent["title"])[0] :
-                (string)torrent["title"];
+            var id = GetFieldAsString(torrent, "identifier");
+            var title = GetFieldAsString(torrent, "title");
+            var btih = GetFieldAsString(torrent, "btih");
 
             release.Title = title;
             release.Comments = new Uri(CommentsUrl + id);
             release.Guid = release.Comments;
 
-            release.PublishDate = DateTime.Now;
-            if (torrent["publicdate"] != null)
-                release.PublishDate = DateTime.Parse((string)torrent["publicdate"]);
+            release.PublishDate = DateTime.Parse(GetFieldAsString(torrent, "publicdate"));
```

Comment:
If you go with the generic version, then I've tested that this line can be simplified to `GetFieldAs<DateTime>(torrent, "publicdate");`

Reply:
done